### PR TITLE
Improvements to test framework cleanup

### PR DESCRIPTION
**Pull request by howardjohn**
* Rename WhenDone to ConditionalCleanup, to avoid ambiguity.
* Allow calling cleanup functions in suite context

This is a pre-cursor PR to one that will make ApplyYAML automatically
cleanup. I may just send that as a followup commit if the reviews are
slow.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Review by lei-tang — APPROVED** [review]
LGTM w.r.t the changes related to security.

**Review by stevenctl — APPROVED** [review]
Just nitpicks, feel free to ignore.

**Review by ramaraochavali — APPROVED** [review]
Looks good apart from already pointed nits

File: pkg/test/framework/config.go [hunk 1]
```diff
@@ -69,31 +65,6 @@ func (c *configManager) ApplyYAMLOrFail(t test.Failer, ns string, yamlText ...st
...
-}
-
-func (c *configManager) ApplyYAMLInClusterOrFail(t test.Failer, cls resource.Cluster, ns string, yamlText ...string) {
-	err := c.ApplyYAMLInCluster(cls, ns, yamlText...)
-	if err != nil {
-		t.Fatal(err)
-	}
-}
-
 func (c *configManager) DeleteYAML(ns string, yamlText ...string) error {
```
[the remark this answers is not in the document]
Comment (howardjohn) [reply pkg/test/framework/config.go:68]: Yeah Ill add a warning

File: pkg/test/framework/components/opentelemetry/kube.go [hunk 2]
```diff
@@ -133,7 +133,7 @@ func install(ctx resource.Context, ns string) error {
 func installServiceEntry(cluster resource.Cluster, ctx resource.Context, ns, ingressAddr string) error {
 	// Setup remote access to zipkin in cluster
 	yaml := strings.ReplaceAll(remoteOtelEntry, "{INGRESS_DOMAIN}", ingressAddr)
-	err := ctx.Config().ApplyYAMLInCluster(cluster, ns, yaml)
```
Comment (howardjohn) [inline pkg/test/framework/components/opentelemetry/kube.go:136]: ApplyYAMLInCluster did NOT actually do what it said. I am going to make it retain the current behvaior (apply to all clusters) and file a bug that the tracing tests are not really multicluster

**Pull request merged**